---
title: "FishGutDADAv2"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r files_and_directories}
# Directories
data.dir = file.path("/data/fishguts_rawdata")
out.dir = file.path("/sharedspace/fish_guts") 
map.path = file.path("/sharedspace/fish_guts/mapping_files")
header_fixed.dir = file.path(out.dir, "header_fixed")
trimmed.dir=file.path("/sharedspace/fish_guts/TrimmedFASTQs")

# make directory for output
dir.create(out.dir, recursive = TRUE)
dir.create(header_fixed.dir, recursive = TRUE)


# Files
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane1.txt")

# Set variables for bash
Sys.setenv(MAP_FILE = map.file)
Sys.setenv(OUT_DIR = out.dir)
Sys.setenv(DATA_DIR = data.dir)
Sys.setenv(HEADER_FIXED_DIR = header_fixed.dir)
Sys.setenv(TRIMMED_DIR = trimmed.dir)
```

```{bash}
cd $DATA_DIR/raw_data
md5sum -c fishgut_rawdata_md5sum.txt
```

```{bash}
cd /sharedspace/fish_guts/DemultiplexedFiles
md5sum * > checklist.txt
```


```{r load_map}
library(readr)
meta.df = read_tsv(map.file)
head(meta.df)
```

```{bash, validate mapping file}
#Be sure correct mapping file is being validated, there are 3 total mapping files. lane 1, 2, 3.
validate_mapping_file.py -m $MAP_FILE -o $OUT_DIR/validate_mapfile
```

```{r, set map file to lane 1 map}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane1.txt")
Sys.setenv(MAP_FILE = map.file)
```

```{bash, combine all R2 and R3 files for pool 1 ran for 150 min}
zcat $DATA_DIR/4_reads/pool_1/*R2*.fastq.gz | \
  sed 's/2:N:0:/1:N:0:/g' | \
  gzip -c > $OUT_DIR/raw_data/R2_combined.fastq.gz

cat $DATA_DIR/4_reads/pool_1/*R3*.fastq.gz > $OUT_DIR/raw_data/R3_combined.fastq.gz

extract_barcodes.py -f $OUT_DIR/raw_data/R2_combined.fastq.gz -r $OUT_DIR/raw_data/R3_combined.fastq.gz -c barcode_paired_end --bc1_len 8 --bc2_len 8 -o $OUT_DIR/barcodes/pool1 -m $MAP_FILE 
```

```{r, set map file to lane 2 map}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane2.txt")
Sys.setenv(MAP_FILE = map.file)
```

```{bash, combine all R2 and R3 files for pool 2}
zcat $DATA_DIR/4_reads/pool_2/*R2*.fastq.gz | \
  sed 's/2:N:0:/1:N:0:/g' | \
  gzip -c > $OUT_DIR/raw_data/R2_combined.fastq.gz

cat $DATA_DIR/4_reads/pool_2/*R3*.fastq.gz > $OUT_DIR/raw_data/R3_combined.fastq.gz

extract_barcodes.py -f $OUT_DIR/raw_data/R2_combined.fastq.gz -r $OUT_DIR/raw_data/R3_combined.fastq.gz -c barcode_paired_end --bc1_len 8 --bc2_len 8 -o $OUT_DIR/barcodes/pool2 -m $MAP_FILE 
```

```{r, set map file to lane 3 map}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane3.txt")
Sys.setenv(MAP_FILE = map.file)
```

```{bash, combine all R2 and R3 files for pool 3}
zcat $DATA_DIR/4_reads/pool_3/*R2*.fastq.gz | \
  sed 's/2:N:0:/1:N:0:/g' | \
  gzip -c > $OUT_DIR/raw_data/R2_combined.fastq.gz

cat $DATA_DIR/4_reads/pool_3/*R3*.fastq.gz > $OUT_DIR/raw_data/R3_combined.fastq.gz

extract_barcodes.py -f $OUT_DIR/raw_data/R2_combined.fastq.gz -r $OUT_DIR/raw_data/R3_combined.fastq.gz -c barcode_paired_end --bc1_len 8 --bc2_len 8 -o $OUT_DIR/barcodes/pool3 -m $MAP_FILE 
```

```{bash, combine R1 and R4 files for pools 1, 2, and 3}
cat $DATA_DIR/4_reads/pool_1/*R1*.fastq.gz >$OUT_DIR/raw_data/R1_combinedpool1.fastq.gz
cat $DATA_DIR/4_reads/pool_1/*R4*.fastq.gz >$OUT_DIR/raw_data/R4_combinedpool1.fastq.gz

cat $DATA_DIR/4_reads/pool_2/*R1*.fastq.gz >$OUT_DIR/raw_data/R1_combinedpool2.fastq.gz
cat $DATA_DIR/4_reads/pool_2/*R4*.fastq.gz >$OUT_DIR/raw_data/R4_combinedpool2.fastq.gz

cat $DATA_DIR/4_reads/pool_3/*R1*.fastq.gz >$OUT_DIR/raw_data/R1_combinedpool3.fastq.gz
cat $DATA_DIR/4_reads/pool_3/*R4*.fastq.gz >$OUT_DIR/raw_data/R4_combinedpool3.fastq.gz
```

```{bash, remove adapters on pool1}
#Need to install cutadapt

cutadapt --match-read-wildcards -g XGTG -g XAGTG -g XTCGTG -g XCACGTG -g XACAAGTG -g XTGCAAGTG -o $TRIMMED_DIR/cleaned.R1.pool1.fq.gz $OUT_DIR/raw_data/R1_combinedpool1.fastq.gz

cutadapt --match-read-wildcards -g XGGAC -g XAGGAC -g XTAGGAC -g XCCCGGAC -g XATTTGGAC -g XGCACAGGAC -o $TRIMMED_DIR/cleaned.R4.pool1.fq.gz $OUT_DIR/raw_data/R4_combinedpool1.fastq.gz
```

```{bash, remove adapters on pool2}
cutadapt --match-read-wildcards -g XGTG -g XAGTG -g XTCGTG -g XCACGTG -g XACAAGTG -g XTGCAAGTG -o $TRIMMED_DIR/cleaned.R1.pool2.fq.gz $OUT_DIR/raw_data/R1_combinedpool2.fastq.gz

cutadapt --match-read-wildcards -g XGGAC -g XAGGAC -g XTAGGAC -g XCCCGGAC -g XATTTGGAC -g XGCACAGGAC -o $TRIMMED_DIR/cleaned.R4.pool2.fq.gz $OUT_DIR/raw_data/R4_combinedpool2.fastq.gz
```


```{bash, remove adapters on pool3}
cutadapt --match-read-wildcards -g XGTG -g XAGTG -g XTCGTG -g XCACGTG -g XACAAGTG -g XTGCAAGTG -o $TRIMMED_DIR/cleaned.R1.pool3.fq.gz $OUT_DIR/raw_data/R1_combinedpool3.fastq.gz

cutadapt --match-read-wildcards -g XGGAC -g XAGGAC -g XTAGGAC -g XCCCGGAC -g XATTTGGAC -g XGCACAGGAC -o $TRIMMED_DIR/cleaned.R4.pool3.fq.gz $OUT_DIR/raw_data/R4_combinedpool3.fastq.gz
```

```{r, set map file to lane 1 map}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane1.txt")
Sys.setenv(MAP_FILE = map.file)
```

```{bash, pool 1 demultiplex}
split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R1.pool1.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool1/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool1/R1 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16

split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R4.pool1.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool1/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool1/R4 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16
```

```{r}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane2.txt")
Sys.setenv(MAP_FILE = map.file)    
```

```{bash}
split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R1.pool2.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool2/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool2/R1 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16

split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R4.pool2.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool2/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool2/R4 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16
```

```{r}
map.file = file.path(map.path,"2015-02-10_AdultStarvation_Map_lane3.txt")
Sys.setenv(MAP_FILE = map.file)
```

```{bash}
split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R1.pool3.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool3/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool3/R1 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16

split_libraries_fastq.py \
    --sequence_read_fps $TRIMMED_DIR/cleaned.R4.pool3.fq.gz \
    --barcode_read_fps $OUT_DIR/barcodes/pool3/barcodes.fastq \
    --output_dir $OUT_DIR/split/pool3/R4 \
    --mapping_fps $MAP_FILE \
    --max_bad_run_length 999 \
    --phred_quality_threshold 0 \
    --sequence_max_n 999 \
    --min_per_read_length_fraction 0.0001 \
    --store_demultiplexed_fastq \
    --barcode_type 16
```

```{bash}
split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool1/R1/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool1F
split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool1/R4/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool1R

split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool2/R1/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool2F
split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool2/R4/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool2R

split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool3/R1/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool3F
split_sequence_file_on_sample_ids.py \
  --input_seqs_fp $OUT_DIR/split/pool3/R4/seqs.fastq \
  --file_type fastq \
  --output_dir $OUT_DIR/sample_files/pool3R
```

```{bash, make directories to combine all files}
mkdir $OUT_DIR/DADA2
mkdir $OUT_DIR/DADA2/Forward
mkdir $OUT_DIR/DADA2/Reverse

cp -R $OUT_DIR/sample_files/pool1F/* $OUT_DIR/DADA2/Forward
cp -R $OUT_DIR/sample_files/pool2F/* $OUT_DIR/DADA2/Forward
cp -R $OUT_DIR/sample_files/pool3F/* $OUT_DIR/DADA2/Forward

cp -R $OUT_DIR/sample_files/pool1R/* $OUT_DIR/DADA2/Reverse
cp -R $OUT_DIR/sample_files/pool2R/* $OUT_DIR/DADA2/Reverse
cp -R $OUT_DIR/sample_files/pool3R/* $OUT_DIR/DADA2/Reverse
```

```{bash}
cp $OUT_DIR/DemultiplexedFiles/F* $OUT_DIR/DADA2/Forward
cp $OUT_DIR/DemultiplexedFiles/R* $OUT_DIR/DADA2/Reverse
```

```{r}
require(dada2)
```

```{bash, make directories to run dada2}
mkdir $OUT_DIR/DADA2/filteredF
mkdir $OUT_DIR/DADA2/filteredR
```

```{r}
# File parsing
pathF = "/sharedspace/fish_guts/DADA2/Forward"
pathR = "/sharedspace/fish_guts/DADA2/Reverse"

fastqFs <- sort(list.files(pathF, pattern="fastq", full.names = TRUE))
fastqRs <- sort(list.files(pathR, pattern="fastq", full.names = TRUE))

sample.namesF <- list.files(pathF, pattern="fastq", full.names = TRUE)
sample.namesR <- list.files(pathR, pattern="fastq", full.names = TRUE)

filtpathF = "/sharedspace/fish_guts/DADA2/filteredF" 
filtpathR = "/sharedspace/fish_guts/DADA2/filteredR" 

filtFs <- file.path(filtpathF, sample.namesF)
filtRs <- file.path(filtpathR, sample.namesR)

if(length(fastqFs) != length(fastqRs)) stop("Forward and reverse files do not match.")
```

```{r, check quality of forward reads}
library(dada2)
plotQualityProfile(pathF[1:2])
```

```{r, check quality of reverse reads}
plotQualityProfile(pathR[1:2])
```

```{r}
# Filtering: THESE PARAMETERS ARENT OPTIMAL FOR ALL DATASETS
require(dada2)
out = filterAndTrim(fastqFs, filtFs, fastqRs, filtRs, maxEE=c(2,2), trimLeft = c(16, 20), truncQ=5, maxN=0, rm.phix=TRUE, compress=TRUE, verbose=TRUE, matchIDs = TRUE, multithread = T)

saveRDS(out, "~/out.RDS")

errF = learnErrors(filtFs, multithread = TRUE)
errR = learnErrors(filtRs, multithread = TRUE)

# Infer sequence variants
sample.names <- sapply(strsplit(basename(filtFs), ".fastq"), `[`, 1) 
sample.namesR <- sapply(strsplit(basename(filtRs), ".fastq"), `[`, 1) 
if(!identical(sample.names, sample.namesR)) stop("Forward and reverse files do not match.")
names(filtFs) <- sample.names
names(filtRs) <- sample.names
mergers <- vector("list", length(sample.namesF))
names(mergers) <- sample.namesR

for(sam in sample.names) {
  cat("Processing:", sam, "\n")
    derepF <- derepFastq(filtFs[[sam]])
    ddF <- dada(derepF, err=errF, multithread=TRUE)
    derepR <- derepFastq(filtRs[[sam]])
    ddR <- dada(derepR, err=errR, multithread=TRUE)
    merger <- mergePairs(ddF, derepF, ddR, derepR)
    mergers[[sam]] <- merger
}

# Construct sequence table and remove chimeras
seqtab <- makeSequenceTable(mergers)

saveRDS(seqtab, "~/seqtab.RDS")
seqtab=readRDS("~/seqtab.RDS")
table(nchar(getSequences(seqtab)))

seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% seq(246,255)]
rm(seqtab)

```

```{bash}
head $HOME/seqs.fasta
```

##Creation of tree, need to run through python 2

```{bash, make a tree with SEPP using gg backbone tree}
wget  "https://raw.github.com/smirarab/sepp-refs/master/gg/sepp-package.tar.bz"

tar xvfj sepp-package.tar.bz

cd sepp-package/sepp

python setup.py config -c

python setup.py install --user
```

```{r}
# Remove chimeras
seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE)

uniques=getUniques(seqtab.nochim)

sequences=names(uniques)

uniquesToFasta(uniques,fout = "~/seqs.fasta", ids=sequences)

```

```{bash}
./sepp-package/run-sepp.sh seqs.fasta Fish
```

```{r}
library(ggtree)
library(ape)

seq.tree = read.tree(file = "/sharedspace/fishgutscode/Fish_placement.tog.relabelled.tre") %>%
  ape::keep.tip(., sequences)
```

```{r}
# Assign taxonomy
taxa <- assignTaxonomy(seqtab.nochim, "/sharedspace/fish_guts/silva_nr_v132_train_set.fa.gz", multithread=TRUE)

#species = addSpecies(taxa, "/data/references/dada/silva_species_assignment_v132.fa.gz")

saveRDS(taxa, "~/taxa.RDS")
taxa=readRDS("~/taxa.RDS")

library(readr)
FishDF <- read_csv("/sharedspace/fish_guts/FishDF.csv")

rownames(FishDF)=FishDF$Info

require(phyloseq)
ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(FishDF), 
               tax_table(taxa),
               phy_tree(seq.tree))

saveRDS(ps, "/sharedspace/fishgutscode/ps.RDS")

ps=readRDS("/sharedspace/fishgutscode/ps.RDS")
```

##Completion of DADA2, now begin making phyloseq objects and remove DEVStarv experiment samples

##Need to review this chunk of code
```{r}
fishy.ps = subset_samples(ps, sample_data(ps)$Group == "Fish")
fishonly.f = filter_taxa(fishy.ps, function(x) mean(x) != 0, prune=TRUE) %>%
  subset_taxa(., Kingdom=="Bacteria") 
rm(fishy.ps)
```

```{r remove outliers from dataset}
library(phyloseq)
#get list of sample names with less than 2000 total reads
Sam=as.list(sample_names(subset_samples(fishonly.f, sample_sums(fishonly.f) <=2000)))

Sam

fishonly.fp = prune_samples(sample_sums(fishonly.f)>=2000, fishonly.f)

```

```{r, create merged phyloseq file}
fishy.r = transform_sample_counts(fishonly.fp, function(x) x / sum(x) )
rm(fishonly.f)

#ordination test
ordinate(fishy.r, "PCoA", "bray") %>%
  plot_ordination(fishy.r, ., type="samples", color="TimeFedStarved", axes = 2:3)
print(p1)
```

```{r, use function to make matrix usable for vegan from phyloseq object}
library(vegan)

vegan_otu <- function(physeq) {
    OTU <- otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU <- t(OTU)
    }
    return(as(OTU, "matrix"))
}
```

```{r}
##merge otu table with sample data
fishdr=as.data.frame(cbind(sample_names(fishy.r),sample_data(fishy.r)$ExptlTimePoint, sample_data(fishy.r)$FedStarved, sample_data(fishy.r)$TimeFedStarved, sample_data(fishy.r)$TankName))
fishdr2 <- fishdr[,-1]
fishsamplenames = as.matrix(fishdr[1])
rownames(fishdr) <- fishsamplenames
colnames(fishdr) = c("SampleNames", "ExptlTimePoint", "FedStarved", "TimeFedStarved", "TankName")

##shows exptltimepoint is signficant for posthoc testing
adonis(otu_table(fishy.r)~FedStarved*TimeFedStarved*ExptlTimePoint, permutations = 999, data=fishdr,strata = fishdr$TankName, parralell = 4)

rich.fish=estimate_richness(otu_table(fishonly.fp), split = TRUE, measures = c("Observed", "Shannon"))
```

#0S
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "0S") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse0S.csv", quote = FALSE)
```
#1S
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1S") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse1S.csv", quote = FALSE)
```
#3S
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3S") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse3S.csv", quote = FALSE)
```
#7S
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7S") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse7S.csv", quote = FALSE)
```
#21S
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21S") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names, TankName)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse21S.csv", quote = FALSE)
```
#1R
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1R") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse1R.csv", quote = FALSE)
```
#3R
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3R") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse3R.csv", quote = FALSE)
```
#7R
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7R") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse7R.csv", quote = FALSE)
```
#21R
```{r}
require(tibble)
B2 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21R") %>%
  filter_taxa(., function(x) mean(x) != 0, prune=TRUE)
lefse = as.data.frame(cbind(t(otu_table(B2, taxa_are_rows = FALSE)), tax_table(B2)))

lefse$Info=paste(lefse$Kingdom, lefse$Phylum, lefse$Class, lefse$Order, lefse$Family, lefse$Genus, sep = "|")
lefse$Kingdom = NULL
lefse$Phylum=NULL
lefse$Class=NULL
lefse$Order=NULL
lefse$Family=NULL
lefse$Genus=NULL

LEFseMeta = sample_data(B2) %>%
  data.frame(.) %>%
  mutate(., names = rownames(.)) %>%
  select(., c(FedStarved,names)) %>%
  as.data.frame(.) %>%
  column_to_rownames(., var = "names") %>%
  t(.)

lefse.out=merge(LEFseMeta,lefse, all=TRUE) %>%
  select(Info, everything())

write.csv(lefse.out, "/sharedspace/fishgutscode/Fishylefse21R.csv", quote = FALSE)
```

```{r}
sample_data(fishy.r)$facet = newphyloPCoA2$facet
sample_data(fishy.r)$facet2 = newphyloPCoA2$facet2


fishdr = as.data.frame(sample_data(fishy.r))
BetaFish=vegdist(vegan_otu(fishy.r), "bray")
fishdr$facet=factor(fishdr$facet, levels = c("0", "1", "3", "7", "21"))
fishdr$facet2=factor(fishdr$facet2, levels = c("Starved","Refed"))
betadisfish=betadisper(BetaFish, group=fishdr$TimeFedStarved, type="centroid")
newphyloPCoA=data.frame(scores(betadisfish,display="sites",choice=1:3),fishdr)
newphyloPCoA2=newphyloPCoA
newphyloPCoA$ExptlTimePoint=NULL
newphyloPCoA$facet=NULL
newphyloPCoA$facet2=NULL



PCoA.plot=ggplot(data = newphyloPCoA2, aes(x = PCoA2, y = PCoA3)) + geom_point(data=newphyloPCoA, alpha=0.05) + geom_point(data = newphyloPCoA2, aes(color = FedStarved), alpha=0.7) + facet_grid(facet2~facet) + theme_bw()+stat_ellipse(data=newphyloPCoA2, aes(group=FedStarved,color=FedStarved),type = "t")+scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f"))+labs(x="PCoA 2 [13%]", y="PCoA 3 [10.4%]")

PCoA.plot

adonis(BetaFish~FedStarved*ExptlTimePoint, permutations = 999, data=newphyloPCoA2, parralell = 4)

library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)

pairwise.adonis2(BetaFish~TimeFedStarved, data=newphyloPCoA2, nperm = 999)

permutest(betadisfish)

plot(betadisfish)
boxplot(betadisfish)
mod.HSD <- TukeyHSD(betadisfish)
mod.HSD
plot(mod.HSD)
```

```{r}

```


```{r}
anosim(BetaFish, fishdr$FedStarved, permutations = 999, parallel = 4)

anosim(BetaFish, fishdr$ExptlTimePoint, permutations = 999, parallel = 4)

anosim(BetaFish, fishdr$TimeFedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "0S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "0S")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
fishy.PD = filter_taxa(fishonly.fp, function(x) sum(x > 10) > (0.05*length(x)), TRUE) %>%
  transform_sample_counts(., function(x) x / sum(x) )

UniFish = ordinate(fishy.PD, method = "PCoA", distance = "unifrac")
UniFish$vectors[,1:4]
fishdr = as.data.frame(sample_data(fishy.PD))

fishdr$facet=factor(fishdr$facet, levels = c("0", "1", "3", "7", "21"))
fishdr$facet2=factor(fishdr$facet2, levels = c("Starved","Refed"))

newphyloUni=data.frame(UniFish$vectors[,1:4],fishdr)
newphyloUni2=newphyloUni
newphyloUni$ExptlTimePoint=NULL
newphyloUni$facet=NULL
newphyloUni$facet2=NULL

PCoA.plot=ggplot(data = newphyloUni2, aes(x = Axis.2, y = Axis.3)) + geom_point(data=newphyloUni, alpha=0.05) + geom_point(data = newphyloUni2, aes(color = FedStarved), alpha=0.7) + facet_grid(facet2~facet) + theme_bw()+stat_ellipse(data=newphyloUni2, aes(group=FedStarved,color=FedStarved),type = "t")+scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f"))+labs(x="PCoA 2 [12.6%]", y="PCoA 3 [10.2%]")

PCoA.plot

unifrac.dist <- UniFrac(fishy.PD, 
                        weighted = T, 
                        normalized = TRUE,  
                        parallel = FALSE, 
                        fast = TRUE)

ps.disper <- betadisper(unifrac.dist, group=fishdr$TimeFedStarved, type="centroid")

plot(ps.disper)
boxplot(ps.disper)
mod.HSD <- TukeyHSD(ps.disper)
mod.HSD
plot(mod.HSD)

permanova <- adonis(unifrac.dist ~ scientific_name, data = metadf)

permanova
```

adonis(BetaFish~FedStarved*ExptlTimePoint, permutations = 999, data=newphyloPCoA2, parralell = 4)

library(devtools)
install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)

pairwise.adonis(BetaFish, fishdr2$V4)

install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

pairwise.adonis2(BetaFish~TimeFedStarved, data=newphyloPCoA2, nperm = 999)


```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "1S")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "7S")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "21S")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```
```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "7S")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "3R")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "7R")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r}
BetaD0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

fishdrD0 = subset(fishdr, fishdr$ExptlTimePoint == "21R")
betadisfishD0=betadisper(BetaD0, group=fishdrD0$FedStarved, type="centroid")
scoreD0=data.frame(scores(betadisfishD0,display="sites",choice=1:3),fishdrD0)

anosim(BetaD0, fishdrD0$FedStarved, permutations = 999, parallel = 4)
```

```{r, alphadiversity plot, keep for pub}
library(RColorBrewer)
fishalpha=diversity(otu_table(fishonly.fp), index = "shannon", MARGIN = 1, base = exp(1))
alphaplot=cbind(sample_data(fishonly.fp), fishalpha)

alphaplot$ExptlTimePoint = factor(alphaplot$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(alphaplot, aes(x=FedStarved, y=fishalpha))+geom_boxplot(aes(color=FedStarved))+facet_grid(~ExptlTimePoint)+scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f"))+theme_bw()
p

alphaplot$logV = log10(alphaplot$fishalpha)

Day0fed=subset(alphaplot, alphaplot$TimeFedStarved == "0Sfed")

shapiro.test(Day0fed$logV)

mean(Day0fed$logV)

Day0starved=subset(alphaplot, alphaplot$TimeFedStarved == "0Sstarved")

shapiro.test(Day0starved$logV)

mean(Day0starved$logV)
mean(Day0fed$logV)

#normalize log transformed value to be equal at Day 0

(0.3642402+0.4299504)/2

x=0.3970953/0.4299504

y=0.3970953/0.3642402

library(dplyr)

FedAlpha=filter(alphaplot, FedStarved == "fed") %>% mutate(logV, logVt=logV*x)
StarvedAlpha = filter(alphaplot, FedStarved == "starved") %>% mutate(logV, logVt=logV*y)

AlphaPlot2=full_join(FedAlpha, StarvedAlpha)

p=ggplot(AlphaPlot2, aes(x=ExptlTimePoint, y=logVt)) + geom_point(aes(color=FedStarved), alpha = 0.3) + scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f")) +stat_summary(aes(y=logVt, group = FedStarved), fun.data = mean_se, geom = "errorbar", width = 0.1)+stat_summary(aes(y=logVt, group = FedStarved, color = FedStarved), fun.y = mean, geom = "line")+theme_bw()+xlab("Experimental Time Point")+ylab("Normalized log Alpha Diversity")

p #anova indicates that there is a signficant difference by condition (fedStarved) and time but not condition x time (p = 0.075)

fit=aov(logVt ~ FedStarved*ExptlTimePoint , data=AlphaPlot2)
plot(fit)
summary(fit)
#anova indicates that there is a signficant difference by condition (fedStarved) and time but not condition x time (p = 0.075)
TukeyHSD(fit)
#signficant difference at 7R p=0.0091635 and 21R p=0.0056947 between fed and starved alpha diversity
```

#caculate faith's PD, may need to install picante
```{r}
#install.packages("picante", repos="http://R-Forge.R-project.org")
library(picante)

sample = vegan_otu(fishy.r)

tree = phy_tree(fishy.r)

pd.result = pd(sample, tree)

pd.result$names = rownames(pd.result)

summary(pd.result$PD)

histogram(pd.result$PD)

alphaplot=cbind(sample_data(fishy.r), pd.result)

alphaplot$ExptlTimePoint = factor(alphaplot$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(alphaplot, aes(x=FedStarved, y=fishalpha))+geom_boxplot(aes(color=FedStarved))+facet_grid(~ExptlTimePoint)+scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f"))+theme_bw()
p

alphaplot$logV = log10(alphaplot$PD)

shapiro.test(alphaplot$logV) #p>0.05

Day0fed=subset(alphaplot, alphaplot$TimeFedStarved == "0Sfed")

Day0starved=subset(alphaplot, alphaplot$TimeFedStarved == "0Sstarved")

mean(Day0fed$logV) #1.291432
mean(Day0starved$logV) #1.154062

#normalize log transformed value to be equal at Day 0

(1.291432+1.154062)/2

x=1.222747/1.291432

y=1.222747/1.154062

FedAlpha=filter(alphaplot, FedStarved == "fed") %>% mutate(logV, logVt=logV*x)
StarvedAlpha = filter(alphaplot, FedStarved == "starved") %>% mutate(logV, logVt=logV*y)

mean(FedAlpha$PD)
mean(StarvedAlpha$PD)

AlphaPlot2=full_join(FedAlpha, StarvedAlpha)

AlphaPlot2 = AlphaPlot2 %>%
  mutate(days = case_when(
    ExptlTimePoint == "0S" ~ "0",
    ExptlTimePoint == "1S" ~ "1",
    ExptlTimePoint == "3S" ~ "3",
    ExptlTimePoint =="7S" ~ "7",
    ExptlTimePoint =="21S" ~ "21",
    ExptlTimePoint =="1R" ~ "22",
    ExptlTimePoint =="3R" ~ "24",
    ExptlTimePoint =="7R" ~ "28",
    ExptlTimePoint =="21R" ~ "42"))

AlphaPlot2$days = as.integer(AlphaPlot2$days)

p=ggplot(AlphaPlot2, aes(x=days, y=logVt)) +
  geom_point(aes(color=FedStarved), alpha = 0.3) +
  scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f")) +
  stat_summary(aes(y=logVt, group = FedStarved), fun.data = mean_se, geom = "errorbar", width = 0.1) +
  stat_summary(aes(y=logVt, group = FedStarved, color = FedStarved), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Experimental Time Point")+
  ylab("Normalized log Alpha Diversity (Faith's PD)")+
  scale_x_continuous("Distance",breaks = c(0,1,3,7,21,22,24,28,42), labels = c("0","1","3","7","21","1","3","7","21"), limits = c(-1,43), expand = c(0,0))
p

fit=aov(logVt ~ FedStarved*ExptlTimePoint , data=AlphaPlot2)
plot(fit)
summary(fit)
#anova indicates that there is a signficant difference by condition (fedStarved) and time, and condition x time 
TukeyHSD(fit)
#signficant difference at 1S, 3S, 7S, 21S, and 21R
```

```{r}
p=ggplot(AlphaPlot2, aes(x=days, y=HAA)) +
  geom_point(aes(color=FedStarved), alpha = 0.3) +
  scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f")) +
  stat_summary(aes(y=HAA, group = FedStarved), fun.data = mean_se, geom = "errorbar", width = 0.1) +
  stat_summary(aes(y=HAA, group = FedStarved, color = FedStarved), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Experimental Time Point")+
  ylab("Height at Anterior of Anal Fin (mm)")+
  scale_x_continuous("Distance",breaks = c(0,1,3,7,21,22,24,28,42), labels = c("0","1","3","7","21","1","3","7","21"), limits = c(-1,43), expand = c(0,0))
p

p=ggplot(AlphaPlot2, aes(x=days, y=SL)) +
  geom_point(aes(color=FedStarved), alpha = 0.3) +
  scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f")) +
  stat_summary(aes(y=SL, group = FedStarved), fun.data = mean_se, geom = "errorbar", width = 0.1) +
  stat_summary(aes(y=SL, group = FedStarved, color = FedStarved), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Experimental Time Point")+
  ylab("Standard Length (mm)")+
  scale_x_continuous("Distance",breaks = c(0,1,3,7,21,22,24,28,42), labels = c("0","1","3","7","21","1","3","7","21"), limits = c(-1,43), expand = c(0,0))
p

fit=aov(HAA ~ FedStarved*ExptlTimePoint , data=AlphaPlot2)
plot(fit)
summary(fit)
#anova indicates that there is a signficant difference by condition (fedStarved) and time, and condition x time 
TukeyHSD(fit) #7S, 21S, 1R, 3R, 7R, 21R

fit=aov(SL ~ FedStarved*ExptlTimePoint , data=AlphaPlot2)
plot(fit)
summary(fit)
#anova indicates that there is a signficant difference by condition (fedStarved) and time, and condition x time 
TukeyHSD(fit) #7S, 21S, 1R, 3R, 7R, 21R


```



```{r}
distance(fishy.r, "bray")
```


```{r, export data for lefse, keep for pub}
write.csv(cbind(t(seqtab.nochim), taxa), "~/LEFsE.csv", quote=FALSE)

LEFsE.OTU=as.data.frame(cbind(t(otu_table(fishy.r, taxa_are_rows=FALSE)), tax_table(fishy.r)))

LEFsE.OTU$X1=NULL
LEFsE.OTU$Info=paste(LEFsE.OTU$Kingdom,LEFsE.OTU$Phylum,LEFsE.OTU$Class,LEFsE.OTU$Order,LEFsE.OTU$Family,LEFsE.OTU$Genus,sep = "|")
LEFsE.OTU$Kingdom=NULL
LEFsE.OTU$Phylum=NULL
LEFsE.OTU$Class=NULL
LEFsE.OTU$Order=NULL
LEFsE.OTU$Family=NULL
LEFsE.OTU$Genus = NULL

fishdata3=as.data.frame(t(sample_data(fishy.r)))
fishdata3$Info=paste(row.names(fishdata3))
fishdata4=fishdata3[c("FedStarved","TimeFedStarved","Description"),]

rm(fishdata3)

LEFsE.out=merge(LEFsE.OTU, fishdata4, all=TRUE)

write.csv(LEFsE.out, "~/LEFsE.csv", quote=FALSE)

```

```{r, test if glom is useful for LEfSe}
genus.glom=tax_glom(fishy.r, "Genus")

LEFsE.OTU2=as.data.frame(cbind(t(otu_table(genus.glom, taxa_are_rows=FALSE)), tax_table(genus.glom)))

LEFsE.OTU2$X1=NULL
LEFsE.OTU2$Info=paste(LEFsE.OTU2$Kingdom,LEFsE.OTU2$Phylum,LEFsE.OTU2$Class,LEFsE.OTU2$Order,LEFsE.OTU2$Family,LEFsE.OTU2$Genus,sep = "|")
LEFsE.OTU2$Kingdom=NULL
LEFsE.OTU2$Phylum=NULL
LEFsE.OTU2$Class=NULL
LEFsE.OTU2$Order=NULL
LEFsE.OTU2$Family=NULL
LEFsE.OTU2$Genus = NULL

fishdata3=as.data.frame(t(sample_data(genus.glom)))
fishdata3$Info=paste(row.names(fishdata3))
fishdata4=fishdata3[c("FedStarved","TimeFedStarved","Description"),]

rm(fishdata3)

LEFsE.out2=merge(LEFsE.OTU2, fishdata4, all=TRUE)

write.csv(LEFsE.out2, "~/LEFsE2.csv", quote=FALSE)
```

```{r}
library(phyloseq)
ps.Genus=subset_taxa(fishy.r, Genus=="Vibrio") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

Vibriodf$ExptlTimePoint = factor(Vibriodf$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(Vibriodf, aes(x=FedStarved, y=Vibrio))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("Vibrio Gut Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.85))
p

ps.Genus=subset_taxa(fishy.r, Genus=="Vibrio") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

#shouldn't use, non-parametric data
fit=aov(Vibrio ~ FedStarved*ExptlTimePoint , data=Vibriodf)
plot(fit)
summary(fit)
#anova indicates time by fedstarved is significant
TukeyHSD(fit)
#indicates significant time point is 21S

library(stats)
TimeFedStarved = factor(Vibriodf$TimeFedStarved)
pairwise.wilcox.test(Vibriodf$Vibrio, TimeFedStarved, p.adj = "BH") #1S, 3S, 21S, 1R, 21R

Vibriodf %>%
  filter(., TimeFedStarved == "21Sstarved") %>%
  summary()


Vibriodf %>%
  filter(., TimeFedStarved == "21Sfed") %>%
  summary()
```

```{r}
ps.Genus=subset_taxa(fishy.r, Genus=="Plesiomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
Plesiomonasdf=Genusdf
Plesiomonasdf$Plesiomonas=Plesiomonasdf$Genussums
Plesiomonasdf$Genussums=NULL

Plesiomonasdf$ExptlTimePoint = factor(Plesiomonasdf$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(Plesiomonasdf, aes(x=FedStarved, y=Plesiomonas))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("Plesiomonas Gut Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.5))
p

ps.Genus=subset_taxa(fishy.r, Genus=="Plesiomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

fit=aov(Vibrio ~ FedStarved*ExptlTimePoint , data=Vibriodf)
plot(fit)
summary(fit)
#anova indicates time by fedstarved is significant
TukeyHSD(fit)
#indicates significant time point is 1S, 3S, 1R, 3R

library(stats)
TimeFedStarved = factor(Vibriodf$TimeFedStarved)
pairwise.wilcox.test(Vibriodf$Vibrio, TimeFedStarved, p.adj = "BH")
```

```{r}
#Create env sample phyloseq object
ps.env = subset_samples(ps, sample_data(ps)$Group == "Environment") %>%
  subset_taxa(., Kingdom=="Bacteria")
ps.env.f = filter_taxa(ps.env, function(x) mean(x) != 0, prune=TRUE)
ps.env.fr  = transform_sample_counts(ps.env.f, function(x) x / sum(x) )
rm(ps.env, ps.env.f)
envDF=as.data.frame(sample_data(ps.env.fr))
```

```{r}
ps.Genus=subset_taxa(ps.env.fr, Genus=="Vibrio") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,envDF,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

Vibriodf$ExptlTimePoint = factor(Vibriodf$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(Vibriodf, aes(x=FedStarved, y=Vibrio))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("Vibrio Env Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.85))
p

ps.Genus=subset_taxa(ps.env.fr, Genus=="Vibrio") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,envDF,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

fit=aov(Vibrio ~ FedStarved*ExptlTimePoint , data=Vibriodf)
plot(fit)
summary(fit)
#anova indicates time by fedstarved is significant
TukeyHSD(fit)
#indicates significant time point is 7S, 21S, & 1R 

library(stats)
TimeFedStarved = factor(Vibriodf$TimeFedStarved)
pairwise.wilcox.test(Vibriodf$Vibrio, TimeFedStarved, p.adj = "BH")
```


#Data for Plesiomonas
```{r}
ps.Genus=subset_taxa(ps.env.fr, Genus=="Plesiomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,envDF,deparse.level = 1)
Plesiomonasdf=Genusdf
Plesiomonasdf$Plesiomonas=Plesiomonasdf$Genussums
Plesiomonasdf$Genussums=NULL

Plesiomonasdf$ExptlTimePoint = factor(Plesiomonasdf$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(Plesiomonasdf, aes(x=FedStarved, y=Plesiomonas))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("Plesiomonas Env Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.5))
p

ps.Genus=subset_taxa(ps.env.fr, Genus=="Plesiomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,envDF,deparse.level = 1)
Vibriodf=Genusdf
Vibriodf$Vibrio=Vibriodf$Genussums
Vibriodf$Genussums=NULL

fit=aov(Vibrio ~ FedStarved*ExptlTimePoint , data=Vibriodf)
plot(fit)
summary(fit)
#anova indicates time by fedstarved is significant
TukeyHSD(fit)
#indicates significant time point is 3S & 3R 

library(stats)
TimeFedStarved = factor(Vibriodf$TimeFedStarved)
pairwise.wilcox.test(Vibriodf$Vibrio, TimeFedStarved, p.adj = "BH")
```

```{r}
ps.Genus=subset_taxa(fishy.r, Genus=="Cetobacterium") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
Cetobacteriumdf=Genusdf
Cetobacteriumdf$Cetobacterium=Cetobacteriumdf$Genussums
Cetobacteriumdf$Genussums=NULL

Cetobacteriumdf$ExptlTimePoint = factor(Cetobacteriumdf$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(Cetobacteriumdf, aes(x=FedStarved, y=Cetobacterium))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("Cetobacterium Gut Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.85))
p
```

```{r}
ps.Genus=subset_taxa(fishy.r, Genus=="ZOR0006") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums = rowSums(vegan.Genus)
Genusdf=cbind(Genussums,newphyloPCoA2,deparse.level = 1)
ZOR0006df=Genusdf
ZOR0006df$Cetobacterium=ZOR0006df$Genussums
ZOR0006df$Genussums=NULL

ZOR0006df$ExptlTimePoint = factor(ZOR0006df$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

p=ggplot(ZOR0006df, aes(x=FedStarved, y=Cetobacterium))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+facet_grid(~ExptlTimePoint)+ylab("ZOR0006 Gut Relative Abundance")+scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+ scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ theme_bw()+coord_cartesian(ylim = c(0, 0.85))
p
```

#Make a heatmap based off lefse results
```{r}
ps.Genus= subset_taxa(fishy.r, Genus=="Crenobacter") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
CrenobacterTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(CrenobacterTFS) = c("TimeFedStarved","Crenobacter")

ps.Genus= subset_taxa(fishy.r, Genus=="Plesiomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
PlesiomonasTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(PlesiomonasTFS) = c("TimeFedStarved","Plesiomonas")

ps.Genus= subset_taxa(fishy.r, Genus=="Glutamicibacter") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
GlutamicibacterTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(GlutamicibacterTFS) = c("TimeFedStarved","Glutamicibacter")

ps.Genus= subset_taxa(fishy.r, Genus=="Exiguobacterium") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
ExiguobacteriumTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(ExiguobacteriumTFS) = c("TimeFedStarved","Exiguobacterium")

ps.Genus= subset_taxa(fishy.r, Genus=="Paraclostridium") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
ParaclostridiumTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(ParaclostridiumTFS) = c("TimeFedStarved","Paraclostridium")

ps.Genus= subset_taxa(fishy.r, Genus=="Halomonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
HalomonasTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(HalomonasTFS) = c("TimeFedStarved","Halomonas")

ps.Genus= subset_taxa(fishy.r, Genus=="Macellibacteroides") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
MacellibacteroidesTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(MacellibacteroidesTFS) = c("TimeFedStarved","Macellibacteroides")

ps.Genus= subset_taxa(fishy.r, Genus=="Aeromonas") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
AeromonasTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(AeromonasTFS) = c("TimeFedStarved","Aeromonas")

ps.Genus= subset_taxa(fishy.r, Genus=="Cetobacterium") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
CetobacteriumTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(CetobacteriumTFS) = c("TimeFedStarved","Cetobacterium")

ps.Genus= subset_taxa(fishy.r, Genus=="ZOR0006") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
ZOR0006TFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(ZOR0006TFS) = c("TimeFedStarved","CK-1C4-19")

ps.Genus= subset_taxa(fishy.r, Genus=="Mycobacterium") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
MycobacteriumTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(MycobacteriumTFS) = c("TimeFedStarved","Mycobacterium")

ps.Genus= subset_taxa(fishy.r, Genus=="Vibrio") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
VibrioTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(VibrioTFS) = c("TimeFedStarved","Vibrio")

ps.Genus= subset_taxa(fishy.r, Genus=="Shewanella") 
vegan.Genus=as.data.frame(vegan_otu(ps.Genus))
Genussums=rowSums(vegan.Genus) 
ShewanellaTFS= cbind(vegan.Genus,alphaplot,deparse.level=1) %>% 
  aggregate(Genussums~TimeFedStarved,.,median) 
colnames(ShewanellaTFS) = c("TimeFedStarved","Shewanella")

MedianGenusDF=as.data.frame(cbind(AeromonasTFS$Aeromonas,CetobacteriumTFS$Cetobacterium,ZOR0006TFS$`CK-1C4-19`, CrenobacterTFS$Crenobacter, ExiguobacteriumTFS$Exiguobacterium, GlutamicibacterTFS$Glutamicibacter, HalomonasTFS$Halomonas, MacellibacteroidesTFS$Macellibacteroides, ParaclostridiumTFS$Paraclostridium, PlesiomonasTFS$Plesiomonas, ShewanellaTFS$Shewanella, VibrioTFS$Vibrio))

rownames(MedianGenusDF)=AeromonasTFS$TimeFedStarved

colnames(MedianGenusDF)=c("Aeromonas", "Cetobacterium", "CK-1C4-19", "Crenobacter", "Exiguobacterium", "Glutamicibacter", "Halomonas", "Macellibacteroides", "Paraclostridium", "Plesiomonas", "Shewanella", "Vibrio")

dfa=MedianGenusDF[c(1,5,13,17,9,3,11,15,7),]
dfb=MedianGenusDF[c(2,6,14,18,10,4,12,16,8),]
DFc <- log2(dfa/dfb)
rownames(DFc)=c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R")

#Remove 0S
DFc = DFc[-1,]

#Change Inf and -Inf values to max and min 
DFc[c(2:4),4]=7.169312
DFc[4,8]=7.169312
DFc[4,10]=7.169312

DFc[5,8]=min(DFc$Vibrio)

genus_matrix=as.matrix(t(DFc))

library(RColorBrewer)
coul = brewer.pal(9, "BrBG")
coul2=colorRampPalette(coul)(50)

heatmap(genus_matrix, Rowv=NULL, Colv=NULL, col = coul2, scale = "column", margins = c(3,1))
genus_matrixR=genus_matrix*(-1)
heatmap(genus_matrixR, Rowv=NULL, Colv=NA, col = coul2, scale = "column", margins = c(3,1))

install.packages("heatmaply")
library(heatmaply)
install.packages("plotly")
library(plotly)
webshot::install_phantomjs()
plotly::orca()
saveRDS(genus_matrixR, "~/genus_matrixR.RDS")
heatmap=heatmaply(genus_matrixR, colors = BrBG, grid_color = "black", Colv=NULL, file="~/LEfSeheatmap.pdf")
heatmap
```


```{r}
rownames(NewGenusDF)=AeromonasTFS$TimeFedStarved

NewGenusDF$FedStarved=c("fed","fed","fed","fed","fed","fed","fed","fed","fed","starved","starved","starved","starved","starved","starved","starved","starved","starved")
NewGenusDF$ExptlTimePoint= c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R","0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R")
NewGenusDF$Days=c(0,1,3,7,21,22,24,28,42,0,1,3,7,21,22,24,28,42)
NewGenusDF$TimeFedStarved=rownames(NewGenusDF)
NewGenusDF$ExptlTimePoint = factor(NewGenusDF$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))
```

```{r}
dfa=EnvGenusDF[c(1,5,13,17,9,3,11,15,7),]
dfb=EnvGenusDF[c(2,6,14,18,10,4,12,16,8),]
DFc <- log2(dfa/dfb)
rownames(DFc)=c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R")
DFc[is.na(DFc)]=0
rownames(DFc)=DFc$X1
DFc$X1=NULL

write.csv(DFc, "~/DFc.csv")

EnvGenusDF[c(1,5,13,17,9,3,11,15,7,2,6,14,18,10,4,12,16,8),]

saveRDS(EnvGenusDF, "/sharedspace/fish_guts/envgenusDF.RDS")

envgenus_matrix=data.matrix(t(envheatmap))
list=colnames(envheatmap)
envlist=paste(list, "env", sep="_")
colnames(envheatmap)=envlist
write.csv(DFc, file="~/envheatmap.csv")

##filter large genus DF
tGenusDF=as.data.frame(t(GenusDF))
tGenusDF$maxval=apply(GenusDF,2,FUN=mean)
tGenusDF$Names=rownames(tGenusDF)
GenusDF.f=filter(tGenusDF,maxval>0.01)


##Make heatmap for Gut Genus samples

genusdf1=MedianGenusDF[1:9,]
genusdf2=MedianGenusDF[10:18,]
genusDF3 <- log2(genusdf1/genusdf2)
rownames(DFc)=c("1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R")


##Change Na's to 0's
genusDF3[is.na(genusDF3)]=0
##Change -Inf to lowest value in dataframe
genusDF3[c(7:9),2]=min(genusDF3$Vibrio)

genusDF3=DFc

write.csv(genusDF3, "~/genusDF3.csv")
rm(genus_matrix)
DFc <- read_csv("~/DFc.csv")
genus_matrix=as.matrix(t(DFc))
DFc$X1=NULL

library(RColorBrewer)
coul = brewer.pal(9, "BrBG")
coul2=colorRampPalette(coul)(50)

heatmap(genus_matrix, Rowv=NULL, Colv=NULL, col = coul2, scale = "column", margins = c(3,1))
genus_matrixR=genus_matrix*(-1)
heatmap(genus_matrixR, Rowv=NULL, Colv=NA, col = coul2, scale = "column", margins = c(3,1))

rownames(genus_matrixR)

rownames(genus_matrixR)=c("Aeromonas", "Exiguobacterium", "Plesiomonas", "Cetobacterium", "Pseudomonas", "Vibrio", "CK-1C4-19", "Mycobacterium", "Macellibacteroides", "Glutamicibacter", "Paraclostridium", "Crenobacter")

install.packages("heatmaply")
library(heatmaply)
install.packages("plotly")
library(plotly)
webshot::install_phantomjs()
plotly::orca()
saveRDS(genus_matrixR, "~/genus_matrixR.RDS")
heatmap=heatmaply(genus_matrixR, colors = BrBG, grid_color = "black", Colv=NULL, file="~/LEfSeheatmap.pdf")
heatmap
```




#PhilR ordination
```{r}
BiocManager::install("philr")
library(philr)
```

```{r}
Philr = fishonly.fp %>%
  #tax_glom(., taxrank = "genus") %>%
  filter_taxa(., function(x) sum(x > 3) > (0.2*length(x)), TRUE) %>%
  transform_sample_counts(., function(x) x+1) 

phy_tree(Philr) = multi2di(phy_tree(Philr))

is.rooted(phy_tree(Philr))

is.binary.tree(phy_tree(Philr))

phy_tree(Philr) <- makeNodeLabel(phy_tree(Philr), method="number", prefix='n')

name.balance(phy_tree(Philr), tax_table(Philr), 'n1')

otu.table <- vegan_otu(otu_table(Philr))

tree <- phy_tree(Philr)
metadata <- sample_data(Philr)
tax <- tax_table(Philr)

gp.philr <- philr(otu.table, tree, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist <- dist(gp.philr, method="euclidean")

gp.pcoa <- ordinate(Philr, 'PCoA', distance=gp.dist)

plot_ordination(Philr, gp.pcoa, type="samples", color = "TimeFedStarved", shape = "FedStarved")
```

```{r}
newphyloPhilR=data.frame(gp.pcoa$vectors[,1:4],fishdr)
newphyloPhilR2=newphyloPhilR
newphyloPhilR$ExptlTimePoint=NULL
newphyloPhilR$facet=NULL
newphyloPhilR$facet2=NULL

PCoA.plot=ggplot(data = newphyloPhilR2, aes(x = Axis.1, y = Axis.2)) + geom_point(data=newphyloPhilR, alpha=0.05) + geom_point(data = newphyloPhilR2, aes(color = FedStarved), alpha=0.7) + facet_grid(facet2~facet) + theme_bw()+stat_ellipse(data=newphyloPhilR2, aes(group=FedStarved,color=FedStarved),type = "t")+scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f"))+labs(x="PCoA 1 [32.7%]", y="PCoA 2 [20%]")

PCoA.plot
```

```{r}
library(glmnet)
glmmod = glmnet(gp.philr, sample_data(Philr)$FedStarved, alpha =1 , family = "binomial")

top.coords <- as.matrix(coefficients(glmmod, s=0.1))
top.coords <- rownames(top.coords)[which(top.coords != 0)]
(top.coords <- top.coords[2:length(top.coords)])

tc.names <- sapply(top.coords, function(x) name.balance(tree, tax, x))

tc.names
```





```{r}
BrayPS = prune_samples(sample_sums(ps)>=2000, ps) %>%
  transform_sample_counts(., function(x) x / sum(x) ) %>%
  subset_taxa(., Kingdom=="Bacteria") 

Bray = distance(BrayPS, "bray", "samples")

BrayM=as.matrix(Bray) %>%
  melt(.) %>%
  rename(., fish_samples = "Var1", tank_samples = "Var2")

head(BrayM)

FedBray = merge(BrayM, fish_tank_data_fed, by = c("fish_samples", "tank_samples"))

StarvedBray = merge(BrayM, fish_tank_data_starved, by = c("fish_samples", "tank_samples"))

BrayDF = rbind(FedBray, StarvedBray)

rm(BrayM, Bray, FedBray, StarvedBray)

BrayDF$ExptlTimePoint = factor(BrayDF$ExptlTimePoint, levels = c("0S", "1S", "3S", "7S", "21S", "1R", "3R", "7R", "21R"))

ggplot(BrayDF, aes(x=TimeFedStarved, y=value))+geom_boxplot(aes(color=FedStarved, fill = FedStarved), lwd = .3)+
  #facet_grid(~ExptlTimePoint)+ylab("Bray-Curtis")+
  scale_color_manual(breaks=c("fed","starved"), values=c("#000000", "#000000"))+
  scale_fill_manual(values=c("#bf812d", "#35978f"), name = "Group", breaks=c("Fed", "Starved"), labels=c("Fed", "Starved"))+ 
  theme_bw()+coord_cartesian(ylim = c(0, 1))
p

BrayDF = BrayDF %>%
  mutate(days = case_when(
    ExptlTimePoint == "0S" ~ "0",
    ExptlTimePoint == "1S" ~ "1",
    ExptlTimePoint == "3S" ~ "3",
    ExptlTimePoint =="7S" ~ "7",
    ExptlTimePoint =="21S" ~ "21",
    ExptlTimePoint =="1R" ~ "22",
    ExptlTimePoint =="3R" ~ "24",
    ExptlTimePoint =="7R" ~ "28",
    ExptlTimePoint =="21R" ~ "42"))

BrayDF$days = as.integer(BrayDF$days)

p=ggplot(BrayDF, aes(x=days, y=value)) +
  geom_point(aes(color=FedStarved), alpha = 0.3) +
  scale_color_manual(breaks=c("fed","starved"),values=c("#bf812d", "#35978f")) +
  stat_summary(aes(y=value, group = FedStarved), fun.data = mean_se, geom = "errorbar", width = 0.1) +
  stat_summary(aes(y=value, group = FedStarved, color = FedStarved), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Experimental Time Point")+
  ylab("Bray-Curtis Distance, Gut and Environment")+
  scale_x_continuous("Distance",breaks = c(0,1,3,7,21,22,24,28,42), labels = c("0","1","3","7","21","1","3","7","21"), limits = c(-1,43), expand = c(0,0))

fit=aov(value ~ FedStarved*ExptlTimePoint , data=BrayDF)
plot(fit)
summary(fit)
#anova indicates that there is a signficant difference by condition x time (p < 0.05)
TukeyHSD(fit) #sig at 1S, 3S, 21S, 1R
```

```{r}
#install.packages("usedist")
library(usedist)

D0 = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "0S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

D0Fed = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "0S" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

D0Starved = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "0S" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(D0, D0Fed$Info, D0Starved$Info) #0.152

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1S" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1S" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) #0.319

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3S" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3S" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) #0.392

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7S" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7S" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) #0.312

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21S") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21S" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21S" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) #0.368
```

```{r}
Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1R" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "1R" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) 

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3R" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "3R" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) 

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7R" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "7R" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info) 

Day = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21R") %>%
  vegan_otu(.) %>%
  vegdist(., "bray") 

DayF = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21R" & sample_data(fishy.r)$FedStarved == "fed") %>%
  sample_data(.) %>%
  as.data.frame(.)

DayS = subset_samples(fishy.r, sample_data(fishy.r)$ExptlTimePoint == "21R" & sample_data(fishy.r)$FedStarved == "starved") %>%
  sample_data(.) %>%
  as.data.frame(.) 
  
dist_between_centroids(Day, DayF$Info, DayS$Info)
#0.309, 0.281, 0.240, 0.313
```

